src/github.com/stellar/gateway/db/repository.go

```diff
@@ -62,3 +63,12 @@ func (r Repository) GetAllowedUserByDomainAndUserId(domain, userId string) (*ent
 		return allowedUser.(*entities.AllowedUser), err
 	}
 }
+
+func (r Repository) GetReceivedPaymentById(id int64) (*entities.ReceivedPayment, error) {
```

Comment:
Any particular reason this is not a pointer receiver?  Having it not be a pointer means you're doing a bunch of copying when making calls.